Classes/Domain/Repository/HaikuRepository.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+declare(strict_types=1);
+
+namespace T3docs\Examples\Domain\Repository;
+
+use TYPO3\CMS\Core\Database\ConnectionPool;
+
+class HaikuRepository
+{
+    public function __construct(
+        protected readonly ConnectionPool $connectionPool,
+    ) {
+    }
+
+    public function findAll(): array
+    {
+        $queryBuilder = $this->connectionPool->getConnectionForTable('pages')->createQueryBuilder();
+        $result = $queryBuilder->select('*')->from('tx_examples_haiku')->executeQuery();
```

Comment:
```
        return $queryBuilder
            ->select('*')
            ->from('tx_examples_haiku')
            ->executeQuery()
            ->fetchAllAssociative;
```

is better readable. Also consider to only select the fields which are needed. This reduces the memory footprint.

Also consider to add a private const with the table name. So it is defined once.

Reply:
I don't like to much chaining, but that is a question of personal taste. I prefer to have named variables inbetween. This also makes x-debugging easier.

Optimizations such as just selecting some fields is beyond scope. This should be an easely understandabele example on how to create a frontend plugin without extbase.